subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/SmokeContinuousIntegrationTest.groovy

```diff
@@ -32,25 +32,85 @@ class SmokeContinuousIntegrationTest extends AbstractContinuousIntegrationTest {
         }
     }
 
+    def "detects changes when no files are in the project"() {
+        given:
+        def markerFile = file("input/marker")
+        buildFile << """
+            task build {
+              def inputFile = file("input/marker")
+              inputs.files inputFile
+              outputs.files "build/marker"
+              doLast {
+                println "exists: " + inputFile.exists()
+              }
+            }
+        """
+
+        when:
+        succeeds("build")
+        then:
+        output.contains "exists: false"
+
+        when:
+        waitBeforeModification(markerFile)
+        markerFile.text = "created"
+        then:
+        succeeds()
+        output.contains "exists: true"
```

Comment:
Shouldn't this be like this?

```suggestion
        when:
        waitBeforeModification(markerFile)
        markerFile.text = "created"
        succeeds()
        then:
        output.contains "exists: true"
```

Do the expectations somehow change because of continuous build? (Sorry, can't check because it's still indexing...)

Reply:
In continuous build, `succeeds()` doesn't trigger a build but waits for a successful build to be triggered by file changes. That is why I put it in the `then:` block. I could rename `succeeds()` to `successfulBuildTriggered()` or something like that, WDYT?

Reply:
Thanks, that's what I expected. I think in that case a rename would be best. It'd also better match the `noBuildTriggered()` method.

Reply:
I added the methods `successfulBuildTriggered()` and `failingBuildTriggered()`, which replace `succeeds()` and `fails()`.